extensions/ql-vscode/src/log-insights/summary-parser.ts

```diff
@@ -0,0 +1,114 @@
+import * as fs from 'fs-extra';
+
+/**
+ * Location information for a single pipeline invocation in the RA.
+ */
+export interface PipelineInfo {
+  startLine: number;
+  raStartLine: number;
+  raEndLine: number;
+}
+
+/**
+ * Location information for a single predicate in the RA.
+ */
+export interface PredicateSymbol {
+  /**
+   * `PipelineInfo` for each iteration. A non-recursive predicate will have a single iteration `0`.
+   */
+  iterations: Record<number, PipelineInfo>;
+}
+
+/**
+ * Location information for the RA from an evaluation log. Line numbers point into the
+ * human-readable log summary.
+ */
+export interface SummarySymbols {
+  predicates: Record<string, PredicateSymbol>;
+}
+
+// Tuple counts for Expr::Expr::getParent#dispred#f0820431#ff@76d6745o:
+const NON_RECURSIVE_TUPLE_COUNT_REGEXP = /^Evaluated relational algebra for predicate (?<predicateName>\S+) with tuple counts:$/;
+// Tuple counts for Expr::Expr::getEnclosingStmt#f0820431#bf@923ddwj9 on iteration 0 running pipeline base:
+const RECURSIVE_TUPLE_COUNT_REGEXP = /^Evaluated relational algebra for predicate (?<predicateName>\S+) on iteration (?<iteration>\d+) running pipeline (?<pipeline>\S+) with tuple counts:$/;
+const RETURN_REGEXP = /^\s*return /;
+
+/**
+ * Parse a human-readable evaluation log summary to find the location of the RA for each pipeline
+ * run.
```

Comment:
Interesting that we now have 2 parser implementations of the evaluation log summary (one for the JSON version that I used for the tree viewer and this one for human-readable). I wonder if it's worth combining them later on.

Reply:
This parser for the human-readable log is really just to figure out where _in the human-readable log_ each relation is. We couldn't really tell that by parsing the JSON version. My plan (and I mention this in a comment somewhere) is to move the code that generates the "symbols" file into the CLI as part of generating the human-readable summary, where we'll already know the location of each relation as we emit it.

Reply:
I internally screamed a little when I read that we were parsing the human-readable log. Having looked at it, it's not actually too bad though, since the scope of what we're parsing is pretty limited and there's a clear path to get rid of that parsing too. 👍🏼

Reply:
I will note that between the time I originally wrote this code a month or so ago, and the time I opened the PR, the human-readable format had changed.